src/lib/vm/vmtropy/mod.rs

```diff
@@ -248,3 +249,46 @@ pub fn mint_record(
 
     mint_credits(owner_address, gates, seed)
 }
+
+/// Matches types of literals (that we know are numbers) and turns them into u128 before trying to downcast to the desired type
+pub fn int_from_output<T: std::convert::TryFrom<u128>>(output: &VariableType) -> Result<T>
+where
+    <T as TryFrom<u128>>::Error: std::fmt::Debug,
+{
+    match output {
+        VariableType::Private(user_input_value_type)
```

Comment:
Not sure why yet, but transitions outputs are collected into `VariableType::Private` even when they are output as public on the Aleo programs. @jrchatruc for visibility.

Reply:
Yeah this is something we still need to implement. In general we're not really handling visibility/encryption when it comes to inputs/outputs that are not records. Let's create a ticket for it to do later